src/idpi/operators/hzerocl.py

```diff
@@ -1,27 +1,37 @@
-"""Algorithm for computation of height of zero degree isotherm.
+"""Algorithm for computation of height of zero degree isotherm."""
+
+# Standard library
+from typing import cast
 
-This is done without extrapolation below model orography.
-"""
 # Third-party
+import numpy as np
 import xarray as xr
 
 # First-party
 from idpi.operators.destagger import destagger
 
 
-def fhzerocl(t: xr.DataArray, hhl: xr.DataArray) -> xr.DataArray:
+def fhzerocl(
+    t: xr.DataArray, hhl: xr.DataArray, extrapolate: bool = False
+) -> xr.DataArray:
     """Height of the zero deg C isotherm in m amsl.
 
     The algorithm searches from the top of the atmosphere downwards.
-    No extrapolation below the earth's surface is done.
 
-    Args:
-        t (xarray.DataArray): air temperature in K
-        hhl (xarray.DataArray): heights of the interfaces between
-            vertical layers in m amsl
+    Parameters
+    ----------
+    t : xr.DataArray
+        Air temperature in K.
+    hhl : xr.DataArray
+        Heights of the interfaces between vertical layers in m amsl.
+    extrapolate : bool, optional
+        Allow the extrapolation of the search below the lowest model layer.
```

Comment:
Perhaps this description of extrapolate can be improved because still not entirely sure why it is needed and what it achieves. Do we have the temperatures below ground in `t`?

Reply:
No the temperature field does not go below the ground. Perhaps there are some soil models that keep track of the underground temperature but those not considered here to my knowledge. Fieldextra proposes an option to extend the gradient of the last 2 model layers below to search for the zero degree C intercept. It's unclear to me whether this information is accurate but it's something that fieldextra does and we need it to reproduce the results of the product.

Reply:
If the lowest model layers reach the surface, is 'below' not then underground? Or do the lowest model layers not necessarily go to the surface? Perhaps my understanding is not 100% necessary, but think the descriptions should be very clear.

Reply:
The lowest model layers do not necessarily reach the surface due to the way that the orography is set up but they may also intersect it. Therefore, the notion of underground depends a bit on the reference. However, the zero degree C intercept may be several kilometers below the lowest model layer and that's definitely underground.

Reply:
Ok interesting, not how I thought the model layers worked.